curated_transformers/tokenizers/tokenizer.py

```diff
@@ -337,8 +338,16 @@ def from_hf_hub_to_cache(
 
     @classmethod
     def from_repo(cls: Type[Self], repo: Repository) -> Self:
+        def validate_tokenizer_file(tokenizer_file: RepositoryFile):
+            # Since repository files are loaded lazily, we need to
+            # open them to check their validity. Will throw if invalid.
+            with tokenizer_file.open(mode="rb") as f:
+                pass
+
         repo = TokenizerRepository(repo)
         tokenizer_file = repo.tokenizer_json()
+        validate_tokenizer_file(tokenizer_file)
```

Comment:
Hmmm, this does make an additional request/has some overhead.

Reply:
Rewrote the initializer to use a single request.